backends-clickhouse/src/main/java/io/glutenproject/vectorized/LowCopyFileSegmentShuffleInputStream.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.vectorized;
```

Comment:
Are the codes totally the same with those in gluten-data? Then maybe we can find out if it's possible to merge them into gluten-core later on.

Reply:
Oh I remember that it's possible but requires for compiling out a minimized utility lib to let ch be to rely on which was determined to be lower-priority. Let's put off doing that for now.